docusaurus/docs/iOS/uikit/controllers/channels.md

```diff
@@ -17,6 +17,23 @@ func controller(
 )
 ```
 
+### Having more than one ChannelListController active at the same time
+
+When having 2 active controllers at the same time, you need to initialize them by passing a `filter` block. This will make sure events are routed to the right list.
+
+```
+client.channelListController(query: query, filter: { channel in
+    // Your filtering logic
+    return true
+})
+```
+
+It is expected that the logic on the `filter` block matches the one in `query.filter`.
+
+:::note
+Given the nature of `Filter`, having generic logic capable of using `query.filter` at runtime to determine if a ChatChannel meets the criteria is extremely expensive and complex. Instead, having a block that only checks for the particular needs of the use case is much cheaper.
+:::
```

Comment:
We don't perform any computations so far so what's written here is actually not true. I also have a strong feeling that non-optional filter would be the way to go in v5 which is on the roadmap. When the filter is required, we won't need expensive `channel.meets(filter)` block. Do we really need this note in place?

Reply:
> having generic logic capable of using `query.filter` at runtime 

The note states that we don't have it, and the reason we don't have it is because it would be expensive. I would leave it as an explanation